In [1]:
import pandas as pd

df = pd.read_csv('./Chỉ tiêu tuyển sinh.csv')

df

CODE                                               NAME  \
0       7520207                        Kỹ thuật Điện tử viễn thông   
1       7520216                 Kỹ thuật Điều khiển và tự động hóa   
2       7510301                    Công nghệ Kỹ thuật Điện-điện tử   
3       7480201                                Công nghệ thông tin   
4       7480202                                  An toàn thông tin   
5       7480101    Khoa học máy tính (định hướng Khoa học dữ liệu)   
6       7480102  Mạng máy tính và truyền thông dữ liệu (CT Kỹ t...   
7   7480201_UDU  Cử nhân Công nghệ thông tin (định hướng ứng dụng)   
8       7329001                           Công nghệ đa phương tiện   
9       7320104                        Truyền thông đa phương tiện   
10      7320101                                            Báo chí   
11      7340101                                Quản trị kinh doanh   
12      7340122                                 Thương mại điện tử   
13      7340115                                          Marketing   
14      7340301                                            Kế toán   
15      7340205                      Công nghệ tài chính (Fintech)   
16  7480201_CLC                                Công nghệ thông tin   
17  7340115_CLC                                          Marketing   
18  7340301_CLC                       Kế toán (chuẩn quốc tế ACCA)   
19      7520207                       Kỹ thuật Điện tử viễn thông    
20      7520216                Kỹ thuật Điều khiển và tự động hóa    
21      7510301                   Công nghệ Kỹ thuật Điện-điện tử    
22      7480201                               Công nghệ thông tin    
23      7480202                                 An toàn thông tin    
24      7520208                         Công nghệ Inernet vạn vật    
25      7329001                          Công nghệ đa phương tiện    
26      7340101                               Quản trị kinh doanh    
27      7340115                                         Marketing    
28      7340301                                           Kế toán    
29  7480201_CLC                               Công nghệ thông tin    
30  7340115_CLC                                         Marketing    

    Xét tuyển tài năng  THPT  XTKH  ĐGNL, ĐGTD  
0                   45   195    95          55  
1                   10    40    20          10  
2                   25   120    60          35  
3                   60   300   150          90  
4                   30   140    70          40  
5                   15    70    35          20  
6                   10    50    25          15  
7                   30   140    70          40  
8                   25   120    60          35  
9                   15    70    35          20  
10                  10    40    20          10  
11                  25   115    55          35  
12                  20    75    35          20  
13                  25   110    55          30  
14                  10    60    30          20  
15                  15    65    30          20  
16                  30   140    70          40  
17                  10    50    25          15  
18                  10    50    25          15  
19                  10    50    30          20  
20                  10    45    25          15  
21                  10    45    25          20  
22                  15    90    50          40  
23                  10    40    25          15  
24                  10    40    25          20  
25                  10    60    35          25  
26                  10    50    30          20  
27                  10    65    35          30  
28                  10    45    25          20  
29                  10    50    30          20  
30                   5    20    10           5

In [5]:
""" 
CODE: Mã ngành xét tuyển	
NAME: Tên ngành
Xét tuyển tài năng: Số chỉ tiêu xét tuyển tài năng
THPT: Số chỉ tiêu thi tốt nghiệp THPT
XTKH: Số chỉ tiêu xét tuyển kết hợp
ĐGNL, ĐGTD: Số chỉ tiêu đánh giá năng lực, đánh giá tư duy
CODE,NAME,THPT,XTKH,DGNL
7520207, Kỹ thuật Điện tử viễn thông, 195, 120, 75
=> - Ngành Kỹ thuật Điện tử viễn thông (Mã ngành: 7520207) tổng chỉ tiêu 390 gồm 15 chỉ tiêu xét tuyển tài năng, 195 chỉ tiêu thi THPT, 120 chỉ tiêu xét tuyển kết hợp, 75 chỉ tiêu đánh giá năng lực, đánh giá tư duy.
 """

df_to_dict = df.to_dict(orient='records')

a = []
for item in df_to_dict:
    index = df_to_dict.index(item)
    if index + 1 < 19:
        str = f"{index+1}. chỉ tiêu tuyển sinh ngành {item['NAME']} tại cơ sở phía Bắc (BVH) (Mã ngành {item['CODE']}) là {item['Xét tuyển tài năng'] + item['THPT'] + item['XTKH'] + item['ĐGNL, ĐGTD']} gồm {item['Xét tuyển tài năng']} chỉ tiêu xét tuyển tài năng, {item['THPT']} chỉ tiêu thi THPT, {item['XTKH']} chỉ tiêu xét tuyển kết hợp, {item['ĐGNL, ĐGTD']} chỉ tiêu đánh giá năng lực, đánh giá tư duy."
        a.append([item['CODE'], str])
    else:
        str = f"{index+1}. chỉ tiêu tuyển sinh ngành {item['NAME']} tại cơ sở phía Nam (BVS) (Mã ngành {item['CODE']}) là {item['Xét tuyển tài năng'] + item['THPT'] + item['XTKH'] + item['ĐGNL, ĐGTD']} gồm {item['Xét tuyển tài năng']} chỉ tiêu xét tuyển tài năng, {item['THPT']} chỉ tiêu thi THPT, {item['XTKH']} chỉ tiêu xét tuyển kết hợp, {item['ĐGNL, ĐGTD']} chỉ tiêu đánh giá năng lực, đánh giá tư duy."
        a.append([item['CODE'], str])

a.sort(key=lambda x: x[0])

for item in a:
    print(item[1])
    

11. chỉ tiêu tuyển sinh ngành Báo chí tại cơ sở phía Bắc (BVH) (Mã ngành 7320101) là 80 gồm 10 chỉ tiêu xét tuyển tài năng, 40 chỉ tiêu thi THPT, 20 chỉ tiêu xét tuyển kết hợp, 10 chỉ tiêu đánh giá năng lực, đánh giá tư duy.
10. chỉ tiêu tuyển sinh ngành Truyền thông đa phương tiện tại cơ sở phía Bắc (BVH) (Mã ngành 7320104) là 140 gồm 15 chỉ tiêu xét tuyển tài năng, 70 chỉ tiêu thi THPT, 35 chỉ tiêu xét tuyển kết hợp, 20 chỉ tiêu đánh giá năng lực, đánh giá tư duy.
9. chỉ tiêu tuyển sinh ngành Công nghệ đa phương tiện tại cơ sở phía Bắc (BVH) (Mã ngành 7329001) là 240 gồm 25 chỉ tiêu xét tuyển tài năng, 120 chỉ tiêu thi THPT, 60 chỉ tiêu xét tuyển kết hợp, 35 chỉ tiêu đánh giá năng lực, đánh giá tư duy.
26. chỉ tiêu tuyển sinh ngành Công nghệ đa phương tiện  tại cơ sở phía Nam (BVS) (Mã ngành 7329001) là 130 gồm 10 chỉ tiêu xét tuyển tài năng, 60 chỉ tiêu thi THPT, 35 chỉ tiêu xét tuyển kết hợp, 25 chỉ tiêu đánh giá năng lực, đánh giá tư duy.
12. chỉ tiêu tuyển sinh ngành Quản trị kin

: 

In [1]:
from getpass import getpass
import os

# from langchain.llms import PromptLayerOpenAI

# PROMPTLAYER_API_KEY = getpass()
# os.environ["PROMPTLAYER_API_KEY"] = PROMPTLAYER_API_KEY

os.environ["OPENAI_API_KEY"] = "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"


1.  chỉ tiêu tuyển sinh ngành Công nghệ Inernet vạn vật tại cơ sở phía Nam (BVS) (Mã ngành 7520208) 
2.  chỉ tiêu tuyển sinh ngành Kỹ thuật Điều khiển và tự động hóa tại cơ sở phía Bắc (BVH) (Mã ngành 7520216) 
3.  chỉ tiêu tuyển sinh ngành Kỹ thuật Điều khiển và tự động hóa tại cơ sở phía Nam (BVS) (Mã ngành 7520216) 


In [2]:
# Index
from langchain_openai import OpenAIEmbeddings
from langchain_community.vectorstores import FAISS

embed_model = OpenAIEmbeddings()
vectorstore = FAISS.load_local(
    "../NewVectorDB/",
    OpenAIEmbeddings(),
    allow_dangerous_deserialization=True,
)


retriever = vectorstore.as_retriever(search_kwargs={"k": 5, "threshold": 0.5})
retriever

VectorStoreRetriever(tags=['FAISS', 'OpenAIEmbeddings'], vectorstore=<langchain_community.vectorstores.faiss.FAISS object at 0x00000218AB1BC650>, search_kwargs={'k': 5, 'threshold': 0.5})

In [3]:
# -*- coding: utf-8 -*-

from langchain.text_splitter import RecursiveCharacterTextSplitter, CharacterTextSplitter
from langchain_community.document_loaders import PyPDFLoader, DirectoryLoader, TextLoader
from langchain_community.vectorstores import FAISS
from langchain_community.document_loaders import TextLoader


from torch import cuda
from langchain.embeddings.huggingface import HuggingFaceEmbeddings
from langchain.embeddings import OpenAIEmbeddings

In [5]:
loader = TextLoader("./Chỉ tiêu tuyển sinh.txt")

documents = loader.load()

text_splitter = RecursiveCharacterTextSplitter(chunk_size=2048, chunk_overlap=256)
chunks = text_splitter.split_documents(documents)
chunks


[Document(page_content='Chỉ tiêu tuyển sinh Học viện Bưu chính Viễn thông năm 2024\nChỉ tiêu tuyển sinh 2024 của Học viện Bưu chính Viễn thông\n\nTổng chỉ tiêu tuyển sinh 2024 là 5.060 cho cả 2 cơ sở đào tạo tại Hà Nội và Tp. HCM, trong đó chỉ tiêu tuyển sinh của 02 Cơ sở đào tạo như sau:\n- CƠ SỞ ĐÀO TẠO PHÍA BẮC (Mã trường: BVH) tổng chỉ tiêu 3900 gồm 1950 cho thi THPT (Trung học phổ thông), 1170 cho xét tuyển kết hợp (XTKH), 780 cho đánh giá năng lực, đánh giá tư duy (ĐGNL, ĐGTD).\n- CƠ SỞ ĐÀO TẠO PHÍA NAM (Mã trường: BVS) tổng chỉ tiêu 1160 gồm 580 cho thi THPT (Trung học phổ thông), 335 cho xét tuyển kết hợp (XTKH), 245 cho đánh giá năng lực, đánh giá tư duy (ĐGNL, ĐGTD).', metadata={'source': './Chỉ tiêu tuyển sinh.txt'}),
 Document(page_content='Chi tiết chỉ tiêu tuyển sinh tại CƠ SỞ ĐÀO TẠO PHÍA BẮC (BVH) của từng ngành/nhóm ngành, theo từng phương thức xét tuyển như sau:\n- Ngành Kỹ thuật Điện tử viễn thông (Mã ngành: 7520207) tổng chỉ tiêu 390 gồm 195 cho thi THPT, 120 cho xé

In [9]:
# Embeding
embed_model = OpenAIEmbeddings()
vectorstore.add_documents(chunks)


['5355a13c-3ee9-480f-af85-31de31892c78',
 'adc62d91-fec2-4a6b-a258-01b3d2afbc1b',
 'b5c0ab97-f80e-4a1f-89ca-68258324bab6',
 'dffefb50-62d8-4a25-ad46-1060fc889f65',
 '0bf443c0-df4c-4c9e-8d67-0c1410685d80']

In [12]:
question = "Chỉ tiêu tuyển sinh công nghệ thông tin"
docs = retriever.get_relevant_documents(question)
docs

[Document(page_content='Chi tiết chỉ tiêu tuyển sinh tại CƠ SỞ ĐÀO TẠO PHÍA NAM (BVS) của từng ngành/nhóm ngành, theo từng phương thức xét tuyển như sau:\n\n- Ngành Kỹ thuật Điện tử viễn thông (Mã ngành: 7520207) tổng chỉ tiêu 100 gồm 50 cho thi THPT, 30 cho xét tuyển kết hợp, 20 cho đánh giá năng lực, đánh giá tư duy.\n\n- Ngành Kỹ thuật Điều khiển và tự động hóa (Mã ngành: 7520216) tổng chỉ tiêu 85 gồm 45 cho thi THPT, 25 cho xét tuyển kết hợp, 15 cho đánh giá năng lực, đánh giá tư duy.\n\n- Ngành Công nghệ Kỹ thuật Điện-điện tử (Mã ngành: 7510301) tổng chỉ tiêu 90 gồm 45 cho thi THPT, 25 cho xét tuyển kết hợp, 20 cho đánh giá năng lực, đánh giá tư duy.\n\n- Ngành Công nghệ thông tin (Mã ngành: 7480201) tổng chỉ tiêu 180 gồm 90 cho thi THPT, 50 cho xét tuyển kết hợp, 40 cho đánh giá năng lực, đánh giá tư duy.\n\n- Ngành An toàn thông tin (Mã ngành: 7480202) tổng chỉ tiêu 80 gồm 40 cho thi THPT, 25 cho xét tuyển kết hợp, 15 cho đánh giá năng lực, đánh giá tư duy.\n\n- Ngành Công nghệ 

In [ ]:
vectorstore.save_local("../NewVectorDB/")